src/Stratis.Features.FederatedPeg/TargetChain/WithdrawalTransactionBuilder.cs

```diff
@@ -25,28 +33,42 @@ public class WithdrawalTransactionBuilder : IWithdrawalTransactionBuilder
         private readonly IFederationWalletManager federationWalletManager;
         private readonly IFederationWalletTransactionHandler federationWalletTransactionHandler;
         private readonly IFederatedPegSettings federatedPegSettings;
+        private readonly IStateRepositoryRoot stateRepositoryRoot;
 
         public WithdrawalTransactionBuilder(
             ILoggerFactory loggerFactory,
             Network network,
             IFederationWalletManager federationWalletManager,
             IFederationWalletTransactionHandler federationWalletTransactionHandler,
-            IFederatedPegSettings federatedPegSettings)
+            IFederatedPegSettings federatedPegSettings,
+            IStateRepositoryRoot stateRepositoryRoot = null)
         {
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.network = network;
             this.federationWalletManager = federationWalletManager;
             this.federationWalletTransactionHandler = federationWalletTransactionHandler;
             this.federatedPegSettings = federatedPegSettings;
+            this.stateRepositoryRoot = stateRepositoryRoot;
         }
 
         /// <inheritdoc />
-        public Transaction BuildWithdrawalTransaction(uint256 depositId, uint blockTime, Recipient recipient)
+        public BuildWithdrawalTransactionResult BuildWithdrawalTransaction(uint256 depositId, uint blockTime, Recipient recipient)
         {
             try
             {
                 this.logger.LogDebug("BuildDeterministicTransaction depositId(opReturnData)={0} recipient.ScriptPubKey={1} recipient.Amount={2}", depositId, recipient.ScriptPubKey, recipient.Amount);
 
+                // Can't send funds to known contract address.
```

Comment:
Can we think of any other reasons something like this might happen based on our consensus rules? I'm having a check now

Reply:
This is the only early check we have now. Ideally it would be possible to exercise the all the rules. The late check should catch everything though.